### PR TITLE
Update dependencies to latest versions for hashfiles

### PR DESCRIPTION
**Pull request by thboop**
This PR:
- [ ] Updates the lockfile version to v2, to match node 16's default npm version
- [ ] Updates the dependencies to their latest versions
- [ ] Configures the linter to be less noisy and less maintenance, as a result of updating the eslint version. Hashfiles is a single file project, we are likely okay to not have heavy linting on it.

### CODE REVIEW

File: src/Misc/expressionFunc/hashFiles/.eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-    "plugins": ["jest", "@typescript-eslint"],
-    "extends": ["plugin:github/es6"],
```
Comment (thboop) [inline src/Misc/expressionFunc/hashFiles/.eslintrc.json:3]: per docs we should pin to [recommended](https://github.com/github/eslint-plugin-github#setup)

File: src/Misc/expressionFunc/hashFiles/.eslintrc.json [hunk 2]
```diff
@@ -17,13 +17,16 @@
       "@typescript-eslint/no-require-imports": "error",
       "@typescript-eslint/array-type": "error",
       "@typescript-eslint/await-thenable": "error",
-      "@typescript-eslint/ban-ts-ignore": "error",
+      "@typescript-eslint/naming-convention": [
```
Comment (thboop) [inline src/Misc/expressionFunc/hashFiles/.eslintrc.json:20]: This replaces

```
      "@typescript-eslint/camelcase": "error",
      "@typescript-eslint/class-name-casing": "error",
```

File: src/Misc/expressionFunc/hashFiles/.eslintrc.json [hunk 3]
```diff
@@ -33,27 +36,26 @@
...
       "@typescript-eslint/promise-function-async": "error",
       "@typescript-eslint/require-array-sort-compare": "error",
       "@typescript-eslint/restrict-plus-operands": "error",
-      "semi": "off",
       "@typescript-eslint/semi": ["error", "never"],
       "@typescript-eslint/type-annotation-spacing": "error",
-      "@typescript-eslint/unbound-method": "error"
+      "@typescript-eslint/unbound-method": "error",
+      "filenames/match-regex" : "off",
+      "github/no-then" : 1, // warning
```
Comment (thboop) [inline src/Misc/expressionFunc/hashFiles/.eslintrc.json:54]: new rules introduced, we fail this one, lets warn for now and if we want to update the code later we can

File: src/Misc/expressionFunc/hashFiles/.eslintrc.json [hunk 4]
```diff
@@ -17,13 +17,16 @@
...
+          "selector": "default",
+          "format": ["camelCase"]
+        }
+      ],
       "camelcase": "off",
-      "@typescript-eslint/camelcase": "error",
-      "@typescript-eslint/class-name-casing": "error",
       "@typescript-eslint/explicit-function-return-type": ["error", {"allowExpressions": true}],
       "@typescript-eslint/func-call-spacing": ["error", "never"],
-      "@typescript-eslint/generic-type-naming": ["error", "^[A-Z][A-Za-z]*$"],
```
Comment (thboop) [inline src/Misc/expressionFunc/hashFiles/.eslintrc.json:26]: I removed some depreciated rules, again this is a one file project, I think we are okay without heavy linting.

```
"@typescript-eslint/generic-type-naming": ["error", "^[A-Z][A-Za-z]*$"],
"@typescript-eslint/no-object-literal-type-assertion": "error",
"@typescript-eslint/prefer-interface": "error",

```

File: src/Misc/expressionFunc/hashFiles/src/hashFiles.ts [hunk 5]
```diff
@@ -1,9 +1,9 @@
-import * as glob from '@actions/glob'
 import * as crypto from 'crypto'
 import * as fs from 'fs'
+import * as glob from '@actions/glob'
```
Comment (thboop) [inline src/Misc/expressionFunc/hashFiles/src/hashFiles.ts:3]: reordered due to new rule expecting imports to be in alphabetical order

File: src/Misc/layoutbin/hashFiles/index.js [hunk 6]
```diff
@@ -1557,12 +1557,12 @@ var __importStar = (this && this.__importStar) || function (mod) {
     return result;
 };
 Object.defineProperty(exports, "__esModule", { value: true });
-const glob = __importStar(__webpack_require__(281));
 const crypto = __importStar(__webpack_require__(417));
 const fs = __importStar(__webpack_require__(747));
+const glob = __importStar(__webpack_require__(281));
```
Comment (thboop) [inline src/Misc/layoutbin/hashFiles/index.js:1562]: no changes besides the order of imports being moved around

**Review by TingluoHuang — APPROVED** [review]

**Pull request merged**